### PR TITLE
style: Add "inherit" style attribute to inherit the current style

### PR DESCRIPTION
**Pull request by johannes-wolf**
#143 @fenjalien Please have a look at this. This is a fix for #143 and #147. I would like to do a release after this merge.

### CODE REVIEW

File: src/styles.typ
Comment (fenjalien) [inline src/styles.typ:1]: I'm not entirely sure how this works, so some comments might be needed. Even so I think this implementation could be better, but if its just to get it to work while the styling is re-worked I'm happy to let it slide...

File: src/styles.typ [hunk 1]
```diff
@@ -23,21 +57,33 @@
   if root != none {
```
Comment (fenjalien) [inline src/styles.typ:57]: Is this loop necssary if `inherit` exists? Or could the logic in `inherit` be put here instead?

File: src/styles.typ [hunk 2]
```diff
@@ -23,21 +57,33 @@
...
   }
   return current
 }
 
+/// Resolve style
+///
+/// - current (style): Current style dictionary
+/// - new (style): Style to merge onto current
+/// - root (string,none): Style root key
+/// - inject (style, none): Inject new style at `root`
```
[the remark this answers is not in the document]
Comment (johannes-wolf) [reply src/styles.typ:74]: It is for libraries for injecting their custom style defaults. Maybe we could only do that once if the key does not exist at all in the style. I will change that.

**Comment by johannes-wolf** [comment]
Should we just close this and come up with a new styling system? Maybe a simpler one? Fixing the current one is extremely frustrating, I am running out of ideas, it always breaks. Maybe we do not need a nested system but can use flat keys, like CSS does? This would simplify things a lot.

**Comment by fenjalien** [comment]
I'm working on a proposal for the new system...

**Pull request closed without merging**